core/src/main/java/org/axonframework/commandhandling/model/inspection/AnnotatedChildEntity.java

```diff
@@ -71,7 +73,9 @@ public void publish(EventMessage<?> msg, P declaringInstance) {
         if (forwardEvents) {
             Iterable<C> targets = eventTargetResolver.apply(msg, declaringInstance);
             if (targets != null) {
-                targets.forEach(target -> this.entityModel.publish(msg, target));
+                Collection<C> copy = new ArrayList<>();
```

Comment:
I realized that the ConcurrentModificationException being thrown is not a problem of Axon Framework but a design problem in the model. If there is need for an child entity to edit the parent's collection you should use a collection with a fail-safe iterator such as ConcurrentHashMap/ConcurrentHashSet/CopyOnWriteArrayList.
Perhaps it would be best to make note of this in the documentation instead.